unitylibs/core/workflow/workflow-upload/action-binder.js

```diff
@@ -0,0 +1,359 @@
+/* eslint-disable max-len */
+/* eslint-disable max-classes-per-file */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-restricted-syntax */
+
+import {
+  unityConfig,
+  getUnityLibs,
+  priorityLoad,
+  createTag,
+  getLocale,
+  getLibs,
+  getHeaders,
+} from '../../../scripts/utils.js';
+
+class ServiceHandler {
+  constructor(renderWidget = false, canvasArea = null, unityEl = null) {
+    this.renderWidget = renderWidget;
+    this.canvasArea = canvasArea;
+    this.unityEl = unityEl;
+  }
+
+  async postCallToService(api, options, errorCallbackOptions = {}, failOnError = true) {
+    const postOpts = {
+      method: 'POST',
+      headers: await getHeaders(unityConfig.apiKey),
+      ...options,
+    };
+    try {
+      const response = await fetch(api, postOpts);
+      if (failOnError && response.status !== 200) throw new Error('Operation failed');
+      if (!failOnError) return response;
+      return await response.json();
+    } catch (err) {
+      this.showErrorToast(errorCallbackOptions);
+      throw new Error('Operation failed');
+    }
+  }
+
+  showErrorToast(errorCallbackOptions) {
+    if (!errorCallbackOptions.errorToastEl) return;
+    const msg = this.unityEl.querySelector(errorCallbackOptions.errorType)?.nextSibling.textContent;
+    this.canvasArea.forEach((element) => {
+      element.style.pointerEvents = 'none';
+      const errorToast = element.querySelector('.alert-holder');
+      const closeBtn = errorToast?.querySelector('.alert-close');
+      if (closeBtn) closeBtn.style.pointerEvents = 'auto';
+      if (!errorToast) return;
```

Comment:
```suggestion
      const errorToast = element.querySelector('.alert-holder');
      if (!errorToast) return;
      const closeBtn = errorToast.querySelector('.alert-close');
      if (closeBtn) closeBtn.style.pointerEvents = 'auto';
```
nit: We can bring up the `errorToast` check so we can avoid the previous optional chaining.